InfluxDB persistence for influxdb versions >= 0.9

theoweiss opened the pull request:
This PR should replace PR https://github.com/openhab/openhab/pull/3322 .
All commits from #3322 are squashed into one.

theoweiss commented:
@teichsta can you please make a brief check if the PR looks good.

teichsta commented on bundles/persistence/org.openhab.persistence.influxdb/pom.xml:
```diff
@@ -12,7 +12,7 @@
 
 	<properties>
-		<bundle.symbolicName>org.openhab.persistence.influxdb</bundle.symbolicName>
+		<bundle.symbolicName>org.openhab.persistence.influxdb1</bundle.symbolicName>
```
revert this change back to .influxdb

teichsta commented on bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java:
```diff
@@ -0,0 +1,582 @@
+    logger.debug("influxdb persistence service activated");
+    disconnect();
+    password = (String) config.get("password");
+    if (StringUtils.isBlank(password)) {
```
statically import isBlank() for the sake of readability

teichsta commented on bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java:
```diff
@@ -0,0 +1,582 @@
+    }
+
+    url = (String) config.get("url");
+    if (StringUtils.isBlank(url)) {
```
statically import isBlank() for the sake of readability

teichsta commented on bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java:
```diff
@@ -0,0 +1,582 @@
+    }
+
+    user = (String) config.get("user");
+    if (StringUtils.isBlank(user)) {
```
statically import isBlank() for the sake of readability

teichsta commented on bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java:
```diff
@@ -0,0 +1,582 @@
+    }
+
+    dbName = (String) config.get("db");
+    if (StringUtils.isBlank(dbName)) {
```
statically import isBlank() for the sake of readability

teichsta commented on bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java:
```diff
@@ -0,0 +1,582 @@
+        }
+      } catch (RuntimeException e) {
+        dbStatus = false;
+        logger.error("database connection failed throwing an exception");
```
maybe add the e.getMessage() to the log statement?

theoweiss replied:
I changed it this way:
```
  logger.error("database connection failed", e);
```

teichsta commented on bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java:
```diff
@@ -0,0 +1,582 @@
+        // may be check for version >= 0.9
+        if (version != null && !version.contains("unknown")) {
+          dbStatus = true;
+          logger.debug("database status is OK");
```
you could also log the version here

teichsta commented on bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java:
```diff
@@ -0,0 +1,582 @@
+          dbStatus = true;
+          logger.debug("database status is OK");
+        } else {
+          logger.error("database connection failed, version is: \"{}\" response time was \"{}\"",
```
is correct to log "connection failed"? At this point we can only say that the ping failed, right? A physical connection could be available though.

teichsta commented on bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java:
```diff
@@ -0,0 +1,582 @@
+      state = item.getStateAs(PercentType.class);
+      logger.trace("percentype item found {}", name);
+    } else {
+      logger.trace("other item found");
```
better unify the different states to be logged here in order to ease debugging. Something linke "Tried to get item as {}"

teichsta commented on bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java:
```diff
@@ -0,0 +1,582 @@
+            .time(System.currentTimeMillis(), timeUnit).build();
+    try {
+      influxDB.write(dbName, "default", point);
+      // influxDB.write(batchPoints);
```
remove commented code

teichsta commented on bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java:
```diff
@@ -0,0 +1,582 @@
+    query.append("select ");
+    query.append(VALUE_COLUMN_NAME);
+    // query.append(", ");
+    // query.append(TIME_COLUMN_NAME);
```
remove commented code

teichsta commented on bundles/persistence/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java:
```diff
@@ -0,0 +1,582 @@
+    logger.trace("filter getBeginDate: {}", filter.getBeginDate());
+    logger.trace("filter getEndDate: {}", filter.getEndDate());
+    logger.trace("filter getPageSize: {}", filter.getPageSize());
+    logger.trace("filter getPageNumber: {}", filter.getPageNumber());
```
better combine these statements into one

teichsta commented:
> @teichsta can you please make a brief check if the PR looks good.

sure, please find my review comments inline. After incorporating the feedback, please squash and merge. Please update the wiki accordingly. It should be clear to the former users that this change is API breaking.

sumnerboy12 commented:
Great work @theoweiss - been waiting for this so I can upgrade my InfluxDB instance!

theoweiss commented:
I incorporated your feedback and squashed the commits.
Now I'm testing the influxdb 0.9 service one more time.

theoweiss commented:
I hesitate to merge this PR because:
it's really annoying but this "order by DESC" bug mentioned here https://github.com/openhab/openhab/issues/2748#issuecomment-153864938 changed, but still exists:
```
> select value from ATestTime where  time < 1451517300s ORDER BY time ASC
name: ATestTime
---------------
time		value
1451517180	1.451517120221e+12
1451517240	1.451517180085e+12

> select value from ATestTime where  time < 1451517300s ORDER BY time DESC
> 
```
I found this issue https://github.com/influxdb/influxdb/issues/4944 but it is still open.
I've deleted all influxdb data files and switched to tsm1 but nothing changed:
```
$ grep tsm /usr/local/etc/influxdb.conf.0.9.theo3
  dir = "/Users/theo/.influxdb/datatsm1"
  engine = "tsm1"
```
Any hints how to solve this?

teichsta commented:
since this is a known (and documented) bug i would tend to merge this PR nevertheless in order to let the community test the new API (implementation). Unfortunately there doesn't seem to be that much drive on the issue. I might help to comment on the orig issue @theoweiss?

theoweiss commented:
@teichsta I also thought about the best solution and came to the same decision: let's merge the PR and have the opportunity to get it fixed by the influxdb people or find a workaround for 1.8.1.
